### PR TITLE
Supporting sidecar view directory

### PR DESCRIPTION
**Pull request by jonspalmer**
### Summary
Simple implementation of option 2 from #331

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -191,6 +191,39 @@ class InlineVariantComponent < ViewComponent::Base
...
+├── ...
+├── test_component.rb
+├── test_component.html.erb
+├── ...
+```
+
+#### Sidecar directory
+
+Alternatively views and other assets can be placed in a sidecar directory with the same name as the component.
+This configuration is particulary useful to organize views along side other assets like javascript and css.
```
Comment (joelhawksley) [inline README.md:213]:
```suggestion
As an alternative, views and other assets can be placed in a sidecar directory with the same name as the component, which can be useful for organizing views alongside other assets like Javascript and CSS.
```

File: test/view_component/view_component_test.rb [hunk 2]
```diff
@@ -437,6 +443,16 @@ def test_raise_error_when_variant_template_file_and_inline_variant_call_exist
     assert_includes error.message, "Template file and inline render method found for variant 'phone' in VariantTemplateAndInlineVariantTemplateComponent."
   end
 
+  def test_raise_error_when_template_file_and_sidecar_directory_template_exist
+    error = assert_raises ViewComponent::TemplateError do
+      with_variant :phone do
+        render_inline(TemplateAndSidecarDirectoryTemplateComponent.new)
+      end
```
Comment (joelhawksley) [inline test/view_component/view_component_test.rb:450]:
```suggestion
      render_inline(TemplateAndSidecarDirectoryTemplateComponent.new)
```

**Comment by helphop** [comment]
I apologize if this is an inappropriate place or time to add my findings.  I have been able to achieve a successful organizing of components with their respected supporting files into one directory by following along this post: [https://fugu.dev/2020-04-16-working-with-view-component-and-stimulus-part-2/](url)

**Comment by jaredcwhite** [comment]
@helphop I think namespacing in that manner seems perfectly reasonable. Probably the only downside is while editing those files you'll end up with lots of "component.rb" or "component.scss", etc. files open, and it might be tricky to switch between them. (Of course I have the same complaint with all the "index.html.erb" files of the world, etc.!)

**Comment by cyupa** [comment]
Yep, I also had to add resolved_paths: ["app/components"] to my webpacker.yml. Should be added to the documentation. May I?

**Comment by joelhawksley** [comment]
@cyupa by all means, please send a PR!

**Comment by cyupa** [comment]
@joelhawksley Thanks for building this great tool and loved your RailsConf 2020 CE presentation - that's what convinced me to try it out and glad to hear it comes to Rails 6.1.

I'm a beginner when it comes to Rails and Stimulus and I found out what the issue was later that night.
It was *obviously* a problem with the "interface behind the keyboard" that was confused about how Stimulus maps namespaced controllers.

Opened this PR to extend the documentation:
https://github.com/github/view_component/pull/481

Thanks!

**Comment by rickychilcott** [comment]
@cyupa - you didn't add the line `resolved_paths: ["app/components"]` to #481, but it appeared that I needed it in order for my static definition to work (i.e. `static targets = ["output"]`). I don't really understand webpack/webpacker enough to know why it's needed, but is there a reason you left it out of #481 after highlighting it here?

Thanks for your comment, by the way, I would have been lost otherwise.

**Comment by cyupa** [comment]
@rickychilcott The pleasure is mine. To be honest, I'm a beginner myself so I don't comprehend how everything works behind the scenes and I happened to stumble upon `resolved_paths: ["app/components"]` to fix it but because I also had issues with the name of the Stimulus controller I assumed it was because of that.

Yesterday, when I submitted the documentation changes I did a check: I removed `resolved_paths: ["app/components"]` stopped the Rails server and restarted it to see if it still worked and it did.

Today I started working on a new component that had to integrate with Stimulus and guess what, the previous one didn't work anymore, so I went straight to my main suspect: `resolved_paths: ["app/components"]`.

After I added it back, it worked.
@joelhawksley I will try to open a pull request for this one as well.

**Comment by rickychilcott** [comment]
Thanks @cyupa and @helphop (for the original write up).

I wonder if your new component had the `static` line in it for stimulus targets. Without `resolved_paths: ["app/components"]` and stimulus worked ok for me as long as I wasn't defining targets (i.e. `static targets = ["output"]`)

**Comment by cyupa** [comment]
Yep, I think we're also using babel and it didn't compiled it into browser-compatible Javascript but webpack had to bundle it in the first place, I guess.

I opened https://github.com/github/view_component/pull/485.

**Pull request merged**